modules/generative-openai/clients/openai.go

```diff
@@ -37,7 +37,11 @@ var compile, _ = regexp.Compile(`{([\w\s]*?)}`)
 
 func buildUrlFn(isLegacy bool, resourceName, deploymentID, baseURL string) (string, error) {
 	if resourceName != "" && deploymentID != "" {
-		host := "https://" + resourceName + ".openai.azure.com"
+		host := baseURL
+		if host == "" || host == "https://api.openai.com" {
+			// Fall back to old assumption
+			host = "https://" + resourceName + ".openai.azure.com"
+		}
 		path := "openai/deployments/" + deploymentID + "/chat/completions"
 		queryParam := "api-version=2023-03-15-preview"
```

Comment:
Is there a specific reason why the queryParam for api-version needs to be hardcoded? Could you not allow the user to specify whichever version they want? This way you do not need to keep maintaining this for deprecated versions.

Reply:
I think it should be configurable so user has more control. If something changes with the new version and there’s a time gap between the phasing out of versions, being able to configure  the api version would allow the user to test with new api version to ensure things are working  correctly. In more enterprise validated applications, this is key.

Reply:
@antas-marcin 
Can we also make the API version configurable when using Azure OpenAI through BaseURL? This would allow developers to update the apiVersion based on the suitable version available to them

Reply:
@pi-curst sure I'll add that possibility

Reply:
@OSS-GR @pi-curst I have added configurable `apiVersion` setting which defaults to `2023-05-15` and also `baseURL` support for `Azure OpenAI` endpoints. Here's a docker image that you can try out:

```
semitechnologies/weaviate:preview-add-support-for-baseurl-in-generative-openai-azure-config-f8935f4
```

Reply:
> @OSS-GR @pi-curst I have added configurable `apiVersion` setting which defaults to `2023-05-15` and also `baseURL` support for `Azure OpenAI` endpoints. Here's a docker image that you can try out:
> 
> ```
> semitechnologies/weaviate:preview-add-support-for-baseurl-in-generative-openai-azure-config-f8935f4
> ```

@antas-marcin This has fixed the issue I was facing with the baseURL. I am able to use the generative-openai module now. I was not sure how to pass the apiVersion variable from the Python v4 client so I was not able to test that.

Reply:
`apiVersion` can be set in `moduleConfig` the same way as you set `baseURL`, but it's great that it works! I will now merge it. Thank you for testing it.